File: CPP/Garage.cpp
// Pre-processors and libraries
#pragma once

#include <string>
#include <iostream>
#include <vector>
#include <typeinfo>
#include "Vehicle.cpp"
#include "Car.cpp"
#include "Motorcycle.cpp"

using namespace std;

// Garage Class
class Garage {
    private:
        // Attributes
        string garage_name;
        int garage_area;
        vector<Vehicle> vehicles;
        
    public:
        /* Constructors */

        // Empty Constructor
        Garage() {
            this->garage_name = "n/a";
            this->garage_area = -1;
        }
        // Constructor
        Garage(string garage_name, int garage_area){
            this->garage_name = garage_name;
            this->garage_area = garage_area;
        }
        /* Methods */

        // Get Methods
        string getGarageName() { return this->garage_name; }
        int getLotCapacity() { return this->garage_area; }

        // Set Methods
        void setGarageName(string garage_name) { this->garage_name = garage_name; }
        void setLotCapacity(int garage_area) { this->garage_area = garage_area; }

        // Garage Methods
        void showVehicleList() {
            int idx = 1;
            for (Vehicle veh : this->vehicles) {
                cout << idx << "." << endl;
                if (typeid(veh) == typeid(Car)) {
                    cout << " Tipe : Mobil" << endl;
                } else if (typeid(veh) == typeid(Car)) {
                    cout << " Tipe : Motor" << endl;
                }

                cout << " " << veh.getLicense() << endl;
                cout << " " << veh.getBrand() << endl;
                cout << " " << veh.getProdYear() << endl;
                cout << " " << veh.getColor() << endl;

                if (typeid(veh) == typeid(Car)) {
                    Car car = static_cast<Car&>(veh);
                    cout << " " << car.getCarDoors() << endl;
                    cout << " " << car.getCarSeats() << endl;
                } else if (typeid(veh) == typeid(Car)) {
                    Motorcycle bike = static_cast<Motorcycle&>(veh);
                    cout << " " << bike.getBikeType() << endl;
                    cout << " " << bike.getTankCapacity() << endl;
                }
            }
        }
        void addVehicle(Vehicle vehicle) {
            this->vehicles.push_back(vehicle);
        }
};

File: Python/Garage.py
from Vehicle import Vehicle # Imports Vehicle Class
from Car import Car # Imports Car Class
from Motorcycle import Motorcycle # Imports Motorcycle Class

class Garage:
    # Attribute Declaration
    __garage_name = ""
    __garage_area = -1
    __vehicle_list = []

    # Constructor
    def __init__(self, garage_name="", garage_area=-1):
        self.__garage_name = garage_name
        self.__garage_area = garage_area = garage_area

    # Getter-Setter
    # Get Methods
    def getBikeType(self):
        return self.__garage_name
    def getTankCapacity(self):
        return self.__garage_area
    # Set Methods
    def setBikeType(self, garage_name):
        self.__garage_name = garage_name
    def setTankCapacity(self, garage_area):
        self.__garage_area = garage_area

    # Methods
    def showVehicleList(self):
        idx = 1
        for veh in self.__vehicle_list:
            print(f"{idx}.")
            if (isinstance(veh, Car)):
                print(" Tipe : Mobil")
            elif (isinstance(veh, Motorcycle)):
                print(" Tipe : Motor")
            
            print(f" {veh.getLicense()}")
            print(f" {veh.getBrand()}")
            print(f" {veh.getProdYear()}")
            print(f" {veh.getColor()}")

            if (isinstance(veh, Car)):
                print(f" {veh.getCarDoors()}")
                print(f" {veh.getCarSeats()}")
            elif (isinstance(veh, Motorcycle)):
                print(f" {veh.getBikeType()}")
                print(f" {veh.getTankCapacity()}")
                
            idx += 1
    def addVehicle(self, vehicle):
        self.__vehicle_list.append(vehicle)